common/lib/protocol-definitions/api-report/protocol-definitions.api.md

```diff
@@ -119,6 +120,7 @@ export interface IConnect {
     supportedFeatures?: Record<string, any>;
     tenantId: string;
     token: string | null;
+    // @deprecated
     versions: string[];
```

Comment:
(mostly thinking aloud): driverVersions is practical way to solve problems we have, but it's not ideal in any form. If/when we decide to completely change protocol, then driverVersion will not help, as likely driver would have to support old and new protocols for very long time (or vice versa - service would need to do the same). So, it feels like we do need to have some protocol version in here. To some extent, driverVersion is extension of that (kind of minor/patch part of that version).

Thoughts?
@GaryWilber - also curious about your thoughts here.

Reply:
Could we still use driverVersion for that scenario? If there was a large protocol change, we would have some minimum required driver that supports that change and the server could have that marked down.
I kind of think of driverVersion as superseding protocolVersion since driverVersion will be usable for protocol version checks too.

Though this becomes trickier if there are multiple drivers involved, so having a separate protocolVersion would be useful in the case.. but there's only 1 for ODSP.

Reply:
I'd say - we can solve it when this problem appears, but back-compat at protocol / document schema levels requires decoupling file format version from code version. It becomes more about capabilities of the code (that ideally, we do not deduce from versions, as features might be behind kill-bit switches and pulled out dynamically)

Reply:
So I would remove the deprecated tag for now.